### ISSUE TITLE
The inferred type of '_sfc_main' cannot be named without a reference to ...

### ISSUE THREAD

**Issue by lichong-a** [qmhc/vite-plugin-dts#195]
### Describe the bug

This looks like the same issue, i can not build success always when I use higher version vue.
#[[The inferred type of '_sfc_main' cannot be named without a reference to](https://github.com/vuejs/core/issues/7038)](ttps://github.com/vuejs/core/issues/7038)

### Reproduction

https://github.com/vuejs/core/issues/7038

### System Info

```shell
node: 18.12.1
plateform: windows11
vite: 4.2.1
vue: 3.2.47
```

### Validations

- [X] Read the [FAQ](https://github.com/qmhc/vite-plugin-dts#faq).
- [X] Check that there isn't [already an issue](https://github.com/qmhc/vite-plugin-dts/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by lichong-a** [comment]
console out
```shell
[vite:dts] Start generate declaration files...
✓ built in 170ms
src/knowledge-example/knowledge-example.vue.ts:5:7 - error TS2742: The inferred type of '_sfc_main' cannot be named without a reference to '.pnpm/@vue+shared@3.2.47/node_modules/@vue/shared'. This is likely not portable. A type annotation is necessary.

5 const _sfc_main = /*#__PURE__*/_defineComponent({
        ~~~~~~~~~

[vite:dts] Declaration files built in 1144ms.
```

**Comment by azat-io** [comment]
The same problem

**Comment by daniel007enirman** [comment]
any solution

**Comment by qmhc** [comment]
Should be fixed in `3.0.0-beta.2`.
